clients/githubrepo/client.go

```diff
@@ -27,19 +27,21 @@ import (
 	"github.com/ossf/scorecard/clients"
 )
 
-const repoFilename = "./githubrepo.tar.gz"
+const repoFilename = "/tmp/%v.%v.githubrepo.tar.gz"
 
 type Client struct {
 	repo       *github.Repository
 	repoClient *github.Client
 	ctx        context.Context
 	owner      string
 	repoName   string
+	tarball    string
```

Comment:
IIUC, the problem currently is when you create multiple instances of `githubrepo.Client` they all write to the same file. Seems like that's what you are trying to get around with the `ReleaseRepo` call.

Instead, we can simply allocate a single file per instance of `githubrepo.Client` and continue its usage as it is today. We can use a randomly generated value to do this. Basically, each instance will own 
`/tmp/githubrepo-randomvalue.tar.gz` and only do its I/O ops on that. Wdyt?

Reply:
> IIUC, the problem currently is when you create multiple instances of `githubrepo.Client` they all write to the same file. Seems like that's what you are trying to get around with the `ReleaseRepo` call.

I implemented `ReleaseRepo` to try to clean up the user's machine and leave as little as possible on disk. `ReleaseRepo` actually does not solve the problem of concurrent runs because if one run deletes the file, the others will fail when reading the file.

> Instead, we can simply allocate a single file per instance of `githubrepo.Client` and continue its usage as it is today. We can use a randomly generated value to do this. Basically, each instance will own
> `/tmp/githubrepo-randomvalue.tar.gz` and only do its I/O ops on that. Wdyt?

I think I'll go with that. So long as we don't want to re-use tarballs across different scorecard runs, it will work. Thanks!

Reply:
Using `ioutil.TempFile` which automatically appends a random number to the filename.